src/moaicore/MOAICCParticle.cpp

```diff
@@ -0,0 +1,214 @@
+	else {
+		float startRadius = particleSystem->mEndRadius + particleSystem->mEndRadiusVariance *  USFloat::Rand(-1.0f, 1.0f);
+		float endRadius = particleSystem->mStartRadius + particleSystem->mStartRadiusVariance *  USFloat::Rand(-1.0f, 1.0f);
+		
+		this->mRadius = startRadius;
+		
+		this->mDeltaRadius = (endRadius - startRadius) / this->mTimeToLive;
+		
+		this->mAngle = a;
+		this->mDegreesPerSecond = D2R * (particleSystem->mRotatePerSecond + particleSystem->mRotatePerSecondVariance * USFloat::Rand(-1.0f, 1.0f));
+		
+		// rotational acceleration
+		this->mRotationalAcceleration = D2R * (particleSystem->mRotationalAcceleration + particleSystem->mRotationalAccelVariance * USFloat::Rand(-1.0f, 1.0f));
+		
+	}
+	
+	// deck index
+	if (particleSystem->mIndexMode == MOAICCParticleSystem::INDEX_MODE_RANDOM) { // random
+		u32 endIdx;
+		u32 startIdx;
+		if (particleSystem->mEndIndex > particleSystem->mStartIndex) {
+			endIdx = particleSystem->mEndIndex;
+			startIdx = particleSystem->mStartIndex;
+		} else {
+			endIdx = particleSystem->mStartIndex;
+			startIdx = particleSystem->mEndIndex;
+		}
+		
+		this->mDeckIndex = (u32) Rand(startIdx, endIdx);
+	} else { // sequential
+		u32 span = (particleSystem->mEndIndex - particleSystem->mStartIndex) + 1;
+		if (particleSystem->mEndIndex > particleSystem->mStartIndex) {
+			this->mDeckIndex = particleSystem->mStartIndex + (particleSystem->mSeqIndex++ % span);
+		} else {
+			this->mDeckIndex = particleSystem->mStartIndex;
+		}
+	}
+	
+}
+
+// move constructor
+MOAICCParticle::MOAICCParticle(MOAICCParticle&& other)
+	:mTimeToLive(other.mTimeToLive),
+	mDeckIndex(other.mDeckIndex),
+	mCurrentPosition(other.mCurrentPosition),
+	mStartPosition(other.mStartPosition),
+	mParticleSize(other.mParticleSize),
+	mDeltaParticleSize(other.mDeltaParticleSize),
+	mParticleRotation(other.mParticleRotation),
+	mDeltaParticleRotation(other.mDeltaParticleRotation),
+	mDirection(other.mDirection),
+	mGravity(other.mGravity),
+	mRadialAcceleration(other.mRadialAcceleration),
+	mTangentialAcceleration(other.mTangentialAcceleration),
+	mAngle(other.mAngle),
+	mDegreesPerSecond(other.mDegreesPerSecond),
+	mRadius(other.mRadius),
+	mDeltaRadius(other.mDeltaRadius),
+	mRotationalAcceleration(other.mRotationalAcceleration)
+{
+	for (int index = 0; index < 4; ++index) {
+		mColor[index] = other.mColor[index];
+		mDeltaColor[index] = other.mDeltaColor[index];
+	}
+	
+	other.mTimeToLive = 0.0f;
+	other.mDeckIndex = 0;
+	other.mCurrentPosition.Init(0.0f, 0.0f);
+	other.mStartPosition.Init(0.0f, 0.0f);
+	other.mParticleSize = 0.0f;
+	other.mDeltaParticleSize = 0.0f;
+	other.mParticleRotation = 0.0f;
+	other.mDeltaParticleRotation = 0.0f;
+	other.mDirection.Init(0.0f, 0.0f);
+	other.mGravity.Init(0.0f, 0.0f);
+	other.mRadialAcceleration = 0.0f;
+	other.mTangentialAcceleration = 0.0f;
+	other.mAngle = 0.0f;
+	other.mDegreesPerSecond = 0.0f;
+	other.mRadius = 0.0f;
+	other.mDeltaRadius = 0.0f;
+	other.mRotationalAcceleration = 0.0f;
```

Comment:
You don't need to zero out things on the other object. That's only necessary when you are storing dangerous things like pointers or resource handles that will get cleaned up by the destructor.